include/dirac_quda.h

```diff
@@ -517,11 +535,11 @@ namespace quda {
     virtual void M(ColorSpinorField &out, const ColorSpinorField &in) const;
     virtual void MdagM(ColorSpinorField &out, const ColorSpinorField &in) const;
 
-    virtual void prepare(ColorSpinorField* &src, ColorSpinorField* &sol,
-			 ColorSpinorField &x, ColorSpinorField &b,
-			 const QudaSolutionType) const;
-    virtual void reconstruct(ColorSpinorField &x, const ColorSpinorField &b,
-			     const QudaSolutionType) const;
+    virtual void prepare(cvector_ref<ColorSpinorField> &out, cvector_ref<ColorSpinorField> &in,
```

Comment:
Convention choice, do we want to copy+paste the doxygen from the parent classic for each derived class? I think it'd be the right idea for actually generating `doxygen`, but it'd get pretty verbose pretty quickly. It's not a hill I'll die on one way or the other.